docs/src/research/installation.md

```diff
@@ -0,0 +1,60 @@
+
+* embed the device specific installation raw disk image in the installer (current way) which results in huge installer image
+* dynamically build the device specific installation according to the user interactive selection
+* download a pre-built device specific raw disk image which could result in huge number of configurations
+* use some combination of generic and specific (kind of a compromise)
+
+None of which seem feasible in the long run. None of which are either Ghaf objective in the long run.
+
+But how to achieve a device specific secure system installation without getting lost with the generic Linux distro requirements?
+
+### Declarative, Non-Interactive Installation
+
+Now that we already version control reference device specific secure system declarations, the question is if we can transfer those into the device installations without asking user too many questions that make the installation unnecessarily difficult to implement? This alone is not a novel idea. Automatic Original Equipment Manufacturer (OEM) installers have been doing this for long. Those are often not declarative but rather scripted guidance to answer questions in generic installers.
+
+In manual installation, the target device specific disk partitioning has been left to the user. Traditionally in generic installers it is also risk management. A user typically might not want her device disk getting wiped out no questions asked. Of course we could let the user know what is about to happen and ask the user for agreement in confirmation before we install fully automatically. Declarative configurations can handle user preferences. If one wants to change something, it can be changed in the declarations, stored and shared via version control. [Also including the declarative partitioning](https://github.com/nix-community/disko) that has been tested from within the Ghaf installer.
+
+So, [if you read the article linked earlier](https://numtide.com/blog/we-dont-need-nixos-cloud-images-anymore-2/), you might have already come to conclusion that a secure, virtualized edge device could be handled similar to cloud images. A simple (even secure) boot and installartion supporting media could provide secure network access to the installation target device and then deploy the specific installation from declarations. In practice, boot the target device with an USB media (or even network boot) that provides SSH access to the device using authorized key. After that, one command can be used to install specific secure system configuration automatically. This is actually already used in [Ghaf updates in development](https://tiiuae.github.io/ghaf/ref_impl/development.html) with `nixos-rebuild ... switch` but cannot yet be used with new/clean systems.
```

Comment:
So, if you read the [article linked earlier](https://numtide.com/blog/we-dont-need-nixos-cloud-images-anymore-2/), you might have already concluded that a secure, virtualized edge device could be handled similarly to Cloud images. A simple (even secure) boot and installation supporting media could provide secure network access to the installation target device and then deploy the specific installation from declarations. In practice, boot the target device with a USB media (or even network boot) that provides SSH access to the device using an authorized key. After that, one command can be used to install specific secure system configurations automatically. This is already used in [Ghaf updates in development](https://tiiuae.github.io/ghaf/ref_impl/development.html) with `nixos-rebuild ... switch` but cannot yet be used with new (clean) systems.

Reply:
Mainly changed as proposed.
I changed the last part to refer to https://github.com/tiiuae/ghaf/pull/340 as it has been now tested to work with Ghaf system clean install.